packages/@aws-cdk/aws-cloudfront/lib/cache-policy.ts

```diff
@@ -0,0 +1,298 @@
+
+  /**
+   * Whether to normalize and include the `Accept-Encoding` header in the cache key whem the `Accept-Encoding` header is 'gzip'.
+   * @default false
+   */
+  readonly enableAcceptEncodingGzip?: boolean;
+}
+
+/**
+ * A Cache Policy configuration.
+ *
+ * @resource AWS::CloudFront::CachePolicy
+ * @experimental
+ */
+export class CachePolicy extends Resource implements ICachePolicy {
+
+  /**
+   * Optimize cache efficiency by minimizing the values that CloudFront includes in the cache key.
+   * Query strings and cookies are not included in the cache key, and only the normalized 'Accept-Encoding' header is included.
+   */
+  public static readonly CACHING_OPTIMIZED = CachePolicy.fromManagedCachePolicy('658327ea-f89d-4fab-a63d-7e88639e58f6');
+  /**
+   * Optimize cache efficiency by minimizing the values that CloudFront includes in the cache key.
+   * Query strings and cookies are not included in the cache key, and only the normalized 'Accept-Encoding' header is included.
+   * Disables cache compression.
+   */
+  public static readonly CACHING_OPTIMIZED_FOR_UNCOMPRESSED_OBJECTS = CachePolicy.fromManagedCachePolicy('b2884449-e4de-46a7-ac36-70bc7f1ddd6d');
+  /** Disables caching. This policy is useful for dynamic content and for requests that are not cacheable. */
+  public static readonly CACHING_DISABLED = CachePolicy.fromManagedCachePolicy('4135ea2d-6df8-44a3-9df3-4b5a84be39ad');
+  /** Designed for use with an origin that is an AWS Elemental MediaPackage endpoint. */
+  public static readonly ELEMENTAL_MEDIA_PACKAGE = CachePolicy.fromManagedCachePolicy('08627262-05a9-4f76-9ded-b50ca2e3a84f');
+
+  /** Imports a Cache Policy from its id. */
+  public static fromCachePolicyId(scope: Construct, id: string, cachePolicyId: string): ICachePolicy {
+    return this.fromCachePolicyAttributes(scope, id, { cachePolicyId });
+  }
+
+  /** Imports a Cache Policy from its attributes. */
+  public static fromCachePolicyAttributes(scope: Construct, id: string, attrs: CachePolicyAttributes): ICachePolicy {
+    return new class extends Resource implements ICachePolicy {
+      public readonly cachePolicyId = attrs.cachePolicyId;
+    }(scope, id);
+  }
+
+  /** Use an existing managed cache policy. */
+  private static fromManagedCachePolicy(managedCachePolicyId: string): ICachePolicy {
+    return new class implements ICachePolicy {
+      public readonly cachePolicyId = managedCachePolicyId;
+    }();
+  }
+
+  public readonly cachePolicyId: string;
+
+  constructor(scope: Construct, id: string, props: CachePolicyProps) {
+    super(scope, id, {
+      physicalName: props.cachePolicyName,
+    });
+
+    const minTtl = (props.minTtl ?? Duration.seconds(0)).toSeconds();
+    const defaultTtl = Math.max((props.defaultTtl ?? Duration.days(1)).toSeconds(), minTtl);
+    const maxTtl = Math.max((props.maxTtl ?? Duration.days(365)).toSeconds(), defaultTtl);
+
+    const resource = new CfnCachePolicy(this, 'Default', {
```

Comment:
We usually use `'Resource'` for this logical ID.

Reply:
We _do_, but _should_ we? I guess this is probably a bigger discussion than this PR, but my impression is that we created `Default` after we'd already had lots of `Resource` constructs, but the general inclination was that the `Default` behavior was better.

Reply:
On second thought, using `Default` makes the IDs in the CloudFormation template "naked" (e.g., 'MyCachePolicy' vs 'MyCachePolicyA3B53D2F'). That's ok, but it's a bit jarring in comparison to everything else. Might hold off on adopting `Default` as the default until we've had some more offline conversation.